trin-core/src/portalnet/types.rs

```diff
@@ -705,4 +609,115 @@ mod test {
         let hex_string = "500F";
         ProtocolId::from_str(hex_string).unwrap();
     }
+
+    // test vectors sourced from
+    // https://github.com/ethereum/portal-network-specs/blob/master/portal-wire-test-vectors.md
+    #[test]
+    fn test_vector_ping() {
+        let enr_seq = 1u64;
+        let data_radius = U256::MAX - U256::from(1);
+        let custom_payload = CustomPayload::new(data_radius.as_ssz_bytes());
+        let expected = "0101000000000000000c000000feffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+        let message = Message::Request(Request::Ping(Ping {
+            enr_seq,
+            custom_payload,
+        }));
+        assert_eq!(hex::encode(message.to_bytes()), expected);
+        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
+        assert_eq!(decoded, message);
+    }
+
+    #[test]
+    fn test_vector_pong() {
+        let enr_seq = 1;
+        let data_radius: U256 = U256::max_value() / 2;
+        let custom_payload = CustomPayload::new(data_radius.as_ssz_bytes());
+        let expected = "0201000000000000000c000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff7f";
+        let message = Message::Response(Response::Pong(Pong {
+            enr_seq,
+            custom_payload,
+        }));
+        assert_eq!(hex::encode(message.to_bytes()), expected);
+        let decoded = Message::from_bytes(message.to_bytes().as_slice()).unwrap();
+        assert_eq!(decoded, message);
+    }
+
+    #[test]
+    fn test_vector_find_nodes() {
+        let distances = vec![256, 255];
+        let expected = "03040000000001ff00";
```

Comment:
Because we have not updated the [union selector to start at 0](https://github.com/ethereum/portal-network-specs/pull/108), the expected value does not match the linked test vector. This affects other test vectors as well. If the intent of this PR is to match the wire protocol test vectors, then we should ensure a match before we merge.

Reply:
Ahh good catch, made the update here